### PR TITLE
Save the vault when saving group filter right after a vault version bump

### LINKED ISSUE #1329: Regression as of 3.0: Group selection in view does not persist after closing
### Version

v3.0

### Source

Google Play

### Vault encryption

Yes (with biometric unlock)

### Device

Pixel 7 and Pixel 6A

### Android version

Android 14 AP1A.240305.019.A1

### ROM

_No response_

### Steps to reproduce

- Upgrade from previous 2.x release on Play Store to 3.0, with pre-existing groups and a group selection saved as default view contents.
- Unlock the vault - entries from all groups are shown.  This was not the default state under 2.x.
- Select only one group.
- Close the app (back gesture to home screen).
- Open the app again and unlock the vault.
- Before 3.0 this would maintain the group selection made above.  Now it goes back to showing all groups' entries.
- Workaround:  Toggle biometric unlock off/on.
- Group selection is again "remembered" between sessions.
- Side-effect:  This also seems to reset the state of last backup taken, bringing up the backup reminder banner even though there haven't been any new entries since last backup.

### What do you expect to happen?

Group selection was expected to be persistent.

### What happens instead?

Group selection is cleared.

### Log

_No response_

### PR DESCRIPTION
**Pull request by alexbakker**
Fixes #1329.

### CODE REVIEW

File: app/src/main/java/com/beemdevelopment/aegis/vault/Vault.java [hunk 1]
```diff
@@ -16,6 +16,9 @@ public class Vault {
     private final UUIDMap<VaultEntry> _entries = new UUIDMap<>();
     private final UUIDMap<VaultGroup> _groups = new UUIDMap<>();
 
+    // Whether we've migrated the group list to the new format while parsing the vault
+    private boolean _groupsMigrated = false;
```
Comment (michaelschattgen) [inline app/src/main/java/com/beemdevelopment/aegis/vault/Vault.java:20]: Could you please change this to _areGroupsMigrated, _groupsNeedsMigration or something similiar?

File: app/src/main/java/com/beemdevelopment/aegis/vault/Vault.java [hunk 2]
```diff
@@ -88,7 +93,15 @@ public static Vault fromJson(JSONObject obj) throws VaultException {
...
-    public void migrateOldGroup(VaultEntry entry) {
+    private void setGroupsMigrated() {
+        _groupsMigrated = true;
+    }
+
+    public boolean getGroupsMigrated() {
+        return _groupsMigrated;
+    }
+
+    public boolean migrateOldGroup(VaultEntry entry) {
```
Comment (michaelschattgen) [inline app/src/main/java/com/beemdevelopment/aegis/vault/Vault.java:104]: Perhaps 'needsGroupMigration' is a better name for this function.

**Review by michaelschattgen — CHANGES_REQUESTED** [review]
Nice, just minor nitpicks on the names.

File: app/src/main/java/com/beemdevelopment/aegis/vault/Vault.java (hunk 1 above)
Comment (alexbakker) [reply app/src/main/java/com/beemdevelopment/aegis/vault/Vault.java]: I've settled on ``isGroupsMigrationFresh``.

File: app/src/main/java/com/beemdevelopment/aegis/vault/Vault.java (hunk 2 above)
Comment (alexbakker) [reply app/src/main/java/com/beemdevelopment/aegis/vault/Vault.java:104]: This function doesn't just check whether a group needs to be migrated, it also actually performs the migration.

**Review by michaelschattgen — APPROVED** [review]

**Pull request merged**